File: easy/goal_parser_interpretation.js
var interpret = function(command) {
    let res = "";
    for(let i = 0;i<command.length;i++){
        if(command[i] === "("){
            if(command[i+1] == ")"){
                i++;
                res = res + "o"
            }
        }
        else if(command[i] === ")"){
            continue
        }
        else{
            res = res + command[i]
        }
    }
    return res
};

File: easy/goalParserInterpretation.java
class Solution {
    public String interpret(String command) {
        int i =0;
        String r = "";
        while(i<command.length()){
            if(command.charAt(i)=='G'){
                r = r+'G';
                i++;
            }
            else if(command.charAt(i)=='(' && command.charAt(i+1)==')'){
                r = r + 'o';
                i+=2;
            }
            else{
                r=r+'a'+'l';
                i+=4;
            }
        }
        return r;
    }
}
